bin/conjur-cli/commands/configuration/apply.rb

```diff
@@ -42,9 +42,19 @@ def call
       private
 
       def server_pid
-        cmd = "ps -ef | grep puma | grep -v grep | grep -v cluster | " \
-              "grep conjur | awk '{print $2}' | tr -d '\n'"
-        stdout, _ = @command_runner.capture2(cmd)
+        cmd ="ps -ef | " +
+          # Filter to only puma processes
+          "grep puma | " +
+          # Filter to only puma process for the Conjur API Server. This tag
+          # is defined in the `config/puma.rb`.
+          "grep '[Conjur API Server]' | " + 
+          # Filter out the grep processes
+          "grep -v grep | " +
+          # Filter out the cluster worker processes
+          "grep -v cluster | " +
+          # Extract the process ID
+          "awk '{print $2}' | tr -d '\n'"
```

Comment:
This didn't work for me both in appliance and in RHEL as the puma service that we're looking for has `[possum]` at the end and not `[Conjur API Server]`:
```
[root@ip-172-31-73-129 ec2-user]# ps -ef | grep puma
conjur       848       1  1 06:20 ?        00:00:03 puma 4.3.8 (tcp://localhost:5000) [possum]
conjur       872       1  0 06:20 ?        00:00:01 puma 5.3.1 (tcp://0.0.0.0:2301,tcp://127.0.0.1) [evoke]
ui           891       1  1 06:20 ?        00:00:03 puma 5.3.1 (tcp://localhost:3000) [ui]
ui          1406     891  0 06:20 ?        00:00:00 puma: cluster worker 0: 891 [ui]
ui          1411     891  0 06:20 ?        00:00:00 puma: cluster worker 1: 891 [ui]
conjur      1436     848  0 06:20 ?        00:00:00 puma: cluster worker 0: 848 [possum]
conjur      1440     848  0 06:20 ?        00:00:00 puma: cluster worker 1: 848 [possum]
conjur      2777     872  0 06:23 ?        00:00:00 puma 5.3.1 (tcp://0.0.0.0:2301,tcp://127.0.0.1) [evoke]
conjur      2781     872  0 06:23 ?        00:00:00 puma 5.3.1 (tcp://0.0.0.0:2301,tcp://127.0.0.1) [evoke]
conjur      2785     872  0 06:23 ?        00:00:00 puma 5.3.1 (tcp://0.0.0.0:2301,tcp://127.0.0.1) [evoke]
conjur      2789     872  0 06:23 ?        00:00:00 puma 5.3.1 (tcp://0.0.0.0:2301,tcp://127.0.0.1) [evoke]
conjur      2793     872  0 06:23 ?        00:00:00 puma 5.3.1 (tcp://0.0.0.0:2301,tcp://127.0.0.1) [evoke]
root        3033    1938  0 06:24 pts/0    00:00:00 grep --color=auto puma
```

Replacing `grep '[Conjur API Server]'` with `grep '\\[possum\\]'` in the code above works for me in both envs.